### ISSUE TITLE
[6.0.0] registerEventHandler causes assertion error

### ISSUE THREAD

**Issue by Hwan-seok** [AgoraIO-Extensions/Agora-Flutter-SDK#797]
**Describe the bug**
calling registerEventHandler and ununregisterEventHandler and again registerEventHandler causes following error

```dart
E/flutter ( 7644): 'package:agora_rtc_engine/src/impl/api_caller.dart': Failed assertion: line 451 pos 12: '_irisCEventHandler != null': is not true.
package:agora_rtc_engine/…/impl/api_caller.dart:451
E/flutter ( 7644): #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
E/flutter ( 7644): #1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
E/flutter ( 7644): #2      _ApiCallExecutorInternal.setupIrisRtcEngineEventHandler
package:agora_rtc_engine/…/impl/api_caller.dart:451
E/flutter ( 7644): #3      _ApiCallExecutor._execute
package:agora_rtc_engine/…/impl/api_caller.dart:211
E/flutter ( 7644): <asynchronous suspension>
```

**To Reproduce**
1. joinChannel
2. registerEventHandler
3. leave Channel
4. ununregisterEventHandler
5. joinchannel
6. registerEventHandler => error occurs

**Expected behavior**
It should work right even with multiple iterations of register - unregister.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: Android Nexus 10
 - OS: Android 31

**Additional context**

On Ipad, it hangs forever instead of error.

**Comment by littleGnAl** [comment]
Hello @Hwan-seok , have you called the `RtcEngine.initialize` first?

**Comment by Hwan-seok** [comment]
Yes but at first creating the engine.
Should I initialize every time before joining the channel?

**Comment by littleGnAl** [comment]
It is most likely you called the `RtcEngine.release` but did not call `RtcEngine.initialize` again.

**Comment by Hwan-seok** [comment]
I have had a release in mine because I called destroy before 6.0.
But the release also has some errors so I eliminated them temporally. Its error was the engine was not initialized(-7). 

To sum up, I did not called release.

**Comment by littleGnAl** [comment]
I found that this is caused by the `unregisterEventHandler `, this issue will happen if there is only one `RtcEngineEventHandler` had been registered and then call `unregisterEventHandler`, say if there is only one `RtcEngineEventHandler` the `RtcEngine.registerEventHandler` and `RtcEngine.unregisterEventHandler` can only be called one time. It had been fixed in my development branch.
https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK/blob/33a620cef881e08e29e0a38393608bc4c5c84ccb/lib/src/impl/agora_rtc_engine_impl.dart#L353

So the `RtcEngine.registerEventHandler ` and `RtcEngine.unregisterEventHandler ` can not be called multiple times at this time, it will be fixed on the next release.

**Comment by Hwan-seok** [comment]
Got it.

I leave this issue for tracking purposes.

**Comment by littleGnAl** [comment]
This should be fixed on `agora_rtc_engine: ^6.1.0`

**Comment by littleGnAl** [comment]
@mdusaama0 
The test case has covered this case, I think you can raise a new issue for your case and put the error log to it.
https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK/blob/5efabbf10c24368f4e20095ac2ae8d3fccb79462/test_shard/integration_test_app/integration_test/rtcengine_smoke_test.dart#L150
